### PR TITLE
Update overview.rst

### PR DESCRIPTION
**Pull request by axelborgegmail**
New Overview file containing the Core principles of Sesam

### CODE REVIEW

**Comment by branislavjenco** [comment]
I fixed some typos and sentence flow, hope that's okay!

File: overview.rst [hunk 1]
```diff
@@ -8,77 +8,197 @@ Overview
...
+
+Master data is any structured data that changes, not just the most central data
+-------------------------------------------------------------------------------
+
+The definition of `master
+data <https://en.wikipedia.org/wiki/Master_data>`_ is "Data that
+provides context to transactional data", but in a data platform
+architecture, the definition of transactional data is somewhat unclear.
+We therefore redefine this to be "Data that provides context to time series
+data". With `time-series, <https://en.wikipedia.org/wiki/Time_series>`_
```
Comment (grove) [inline overview.rst:36]: remove dash from "time-series"?

File: overview.rst [hunk 2]
```diff
@@ -8,77 +8,197 @@ Overview
...
+<https://en.wikipedia.org/wiki/Extract,_transform,_load>`_ to retrieve
+master data as opposed to time series data, which in most cases is
+streamed into the platform. A reason for this is that the amount of
+master data is usually limited, and that the source systems often cannot
+deliver a stream of changes. The biggest problem in this approach is
+that master data is always composed from multiple systems, so all data
+from all systems must be reloaded each time data is updated. This causes
+a low refresh rate, and undesired dependencies between source systems to
+make ETL jobs able to complete. This causes all downstream systems that
+need master data to also be forced to batch process therr data using
```
Comment (grove) [inline overview.rst:172]: typo: therre

File: overview.rst [hunk 3]
```diff
@@ -8,77 +8,197 @@ Overview
...
+needed.
+
+Implementation details
+----------------------
+
+1. Sesam should pull data in, and push data out, if possible
+
+2. All data in Sesam must have namespaced properties and IDs
+
+3. All references between data objects should be stored as NIs
```
Comment (TarjeiSkrede) [inline overview.rst:191]: Should use the full representation here, change 'NIs' to 'namespaced identifiers (NIs)'.
NIs is not a defined term yet.

File: overview.rst [hunk 4]
```diff
@@ -8,77 +8,197 @@ Overview
...
+--------------------------------------------------------------------------------
+
+The biggest challenge for data quality is that each system in a
+composite architecture has different representations of the same object.
+The challenge is amplified in a data-driven architecture, where the data
+itself should be meaningful and not wrapped in functional shells as in a
+traditional `SOA
+architecture. <https://en.wikipedia.org/wiki/Service-oriented_architecture>`_
+
+The "simple" solution is to agree on one a single data model for each
```
Comment (TarjeiSkrede) [inline overview.rst:64]: I think these long sections should be split up into smaller sections, which makes them easier to read and so easier to understand. This applies to other sections here as well.

File: overview.rst [hunk 5]
```diff
@@ -8,77 +8,197 @@ Overview
...
+out to all systems that need the change. Regardless of whether a
+source supports change tracking, Sesam will immediately convert any
+batch load to a stream of the real changes contained in that batch using
+delta comparison. Sesam will automatically interpret the dataflow
+configuration so that it knows every single object affected by any
+change. This is a prerequisite for being able to stream compound objects
+without having to ETL load the entire dataset each time updated data is
+needed.
+
+Implementation details
```
Comment (BaardBouvet) [inline overview.rst:184]: I think this belongs in best practices/development guide. Not in the overview.

File: overview.rst [hunk 6]
```diff
@@ -8,77 +8,197 @@ Overview
...
+   outside the system where it was created.*
+
+This means that master data is not limited to customer, product, asset,
+and employee data, but contains both `reference
+data <https://en.wikipedia.org/wiki/Reference_data>`_ and all possible
+forms of structured data stored in a system, where the data also is
+required to be made available outside of that system. Master data is all
+data in a data platform that is neither unstructured nor in time
+series. Examples of master data slightly outside the usual definition
+are: order and invoice information, work orders, hourly, document
```
Comment (TarjeiSkrede) [inline overview.rst:51]: Dumb question, but by 'hourly' do you mean 'hourly data'? And doesn't these come in under the term time series data?

File: overview.rst [hunk 7]
```diff
@@ -8,77 +8,197 @@ Overview
...
+master data as opposed to time series data, which in most cases is
+streamed into the platform. A reason for this is that the amount of
+master data is usually limited, and that the source systems often cannot
+deliver a stream of changes. The biggest problem in this approach is
+that master data is always composed from multiple systems, so all data
+from all systems must be reloaded each time data is updated. This causes
+a low refresh rate, and undesired dependencies between source systems to
+make ETL jobs able to complete. This causes all downstream systems that
+need master data to also be forced to batch process therr data using
+ETL. Sesam is an at-design-time dataflow tool optimized to always just
```
Comment (grove) [inline overview.rst:173]: Add link for "dataflow" to https://en.wikipedia.org/wiki/Dataflow

File: overview.rst [hunk 8]
```diff
@@ -8,77 +8,197 @@ Overview
...
+dependency. Sesam is optimized for synchronizing master data between the
+master data hub and each service's data store in the same way as against
+business systems. Regardless of the optimal store for a service, whether
+it's SQL-based, search-index-based, NoSQL-based, graph-based, or using
+special tools such as Firebase, Qlik, Tableau, etc.
+
+Stream changes to master data, don't use slow and resource intensive ETL
+------------------------------------------------------------------------
+
+Traditional data platform architecture is often based on `ETL
```
Comment (BaardBouvet) [inline overview.rst:162]: Write that this requires dependency tracking and this is hard to implement without bugs when you have complex hops/merges. Kafka Streams also has this, but not with complex hops with multi valued join keys.

File: overview.rst (hunk 6 above)
Comment (axelborgegmail) [reply overview.rst:51]: Sorry, I ment "time registrering" but dint find the right word. I think "Timesheet" will do it

File: overview.rst (hunk 3 above)
Comment (axelborgegmail) [reply overview.rst:191]: Yes, and have a link to the right section in the docs

**Comment by grove** [comment]
Note: I fixed the document title as the formatting was incorrect and the docs could not be built.

**Comment by grove** [comment]
We should also rename `overview.rst` to `core-principles.rst` I think. Agree?

**Comment by grove** [comment]
FYI: I've moved the core principles to core-principle. And I've added back a minimal overview document. Feel free to rewrite and restructure the overview.

File: core-principles.rst [hunk 9]
```diff
@@ -56,12 +55,21 @@ architecture. <https://en.wikipedia.org/wiki/Service-oriented_architecture>`_
...
-represents everything you will ever need, in all systems, forever. This
-is a utopia on which many businesses have wasted enormous resources,
+model <https://en.wikipedia.org/wiki/Canonical_model>`_, that ideally
+represents a standardized superset everything you will ever need, from 
+all systems. 
+While this may be a requirement in a message-based integration
+architecture, and is a truly valubale in a industry-wide data 
+`interoperability <https://en.wikipedia.org/wiki/Interoperability>`_ 
+scenario, in a data platform the canonical model becomes a constant burden 
+demanding continual enhansement and expansion over time. The result becomes a
```
Comment (grove) [inline core-principles.rst:65]: typo: enhansement

File: core-principles.rst [hunk 10]
```diff
@@ -56,12 +55,21 @@ architecture. <https://en.wikipedia.org/wiki/Service-oriented_architecture>`_
...
-is a utopia on which many businesses have wasted enormous resources,
+model <https://en.wikipedia.org/wiki/Canonical_model>`_, that ideally
+represents a standardized superset everything you will ever need, from 
+all systems. 
+While this may be a requirement in a message-based integration
+architecture, and is a truly valubale in a industry-wide data 
+`interoperability <https://en.wikipedia.org/wiki/Interoperability>`_ 
+scenario, in a data platform the canonical model becomes a constant burden 
+demanding continual enhansement and expansion over time. The result becomes a 
+datamodel that fits none of the source systems, and allways lags behind
```
Comment (grove) [inline core-principles.rst:66]: typo: allways

File: core-principles.rst [hunk 11]
```diff
@@ -96,25 +106,29 @@ The master data
...
-system communicates via REST API, SOAP, XML, CSV, SQL or any other
-communication form or format is irelevant. If there is structured
-data, Sesam will be able to receive and return data in the subject
-system's own format and data model. 
+Sesam does not require any modifications in the business systems, but 
+instead each system talk in their own language, meaning they can collect and 
+share data in the form that the system supports, through mechanisms for 
+which the systems were built. Whether the business system communicates via 
+REST API, SOAP, XML, CSV, SQL or any other communication form or format 
+is irelevant. If there is structured data, Sesam will be able to talk to
```
Comment (grove) [inline core-principles.rst:120]: typo: irelevant

File: core-principles.rst [hunk 12]
```diff
@@ -96,25 +106,29 @@ The master data
...
+every system in their own language. 
 
-In Sesam all data is transformed into a JSON based
+Internaly in Sesam all data is transformed into a
```
Comment (grove) [inline core-principles.rst:123]: typo: internaly

Comment (axelborgegmail) [reply core-principles.rst:123]: Sorry! Forgot the spell checker 😳

**Review by grove — APPROVED** [review]

**Pull request merged**